AudioPlayerRecorderSamples/src/main/res/values/strings.xml

```diff
@@ -5,4 +5,8 @@
     <string name="hello_world">Hello world!</string>
     <string name="action_settings">Settings</string>
 
+    <string name="permission_rationale">You need to grant application record audio permission in Settings application.</string>
+    <string name="permission_action_allow">Allow</string>
+    <string name="permission_action_settings">Settings</string>
+    <string name="error_storage_not_available">Storage not available.</string>
```

Comment:
Following [Android's permissions' samples](https://github.com/googlesamples/android-RuntimePermissions), I would make a couple of changes to these strings:
1. Rename `permission_rationale` to `permission_microphone_rationale` in the first 3.
2. Use a different copy, such as: "Microphone permission is needed to record audio." (although they also use "OK" instead of "Settings" for the button, the design spec uses "Settings" and I think it's much clearer, so I think we should keep it).

Also, good catch that storage might not be available!